### PR TITLE
Improve performance to "rit create formula" command

### PR DESCRIPTION
**Pull request by kaduartur**
### Description
Some versions ago we removed the building code of the `rit create formula` command because it is already present at the time of executing a formula, then we realized that we could remove the spinner of the `rit create formula` command. The command now executes instantly when called.

When creating the formula we now generate your hash to get to know if we had any changes to the formula before the first build was done.

### How to verify it
Follow the steps below:
- create a new formula
- edit this formula

The formula changes must be present in the first build.

### Changelog
Remove spinner from `rit create formula`

### CODE REVIEW

File: internal/mocks/mocks.go [hunk 1]
```diff
@@ -146,6 +146,59 @@ func (i *InputTextValidatorMock) Text(name string, validate func(interface{}) er
...
+	return args.String(0), args.Error(1)
+}
+
+func (w *WorkspaceForm) PreviousHash(formulaPath string) (string, error) {
+	args := w.Called(formulaPath)
+	return args.String(0), args.Error(1)
+}
+
+func (w *WorkspaceForm) UpdateHash(formulaPath string, hash string) error {
+	args := w.Called(formulaPath)
```
Comment (brunasilvazup) [inline internal/mocks/mocks.go:198]: In this you would pass the formulaPath and hash normally, do you have a any motivation to pass only one?

**Review by brunasilvazup — APPROVED** [review]

File: internal/mocks/mocks.go [hunk 2]
```diff
@@ -267,3 +330,25 @@ func (t *TreeManager) Check() []api.CommandID {
...
+
+func (tm *TemplateManagerMock) LangTemplateFiles(lang string) ([]template.File, error) {
+	args := tm.Called(lang)
+	return args.Get(0).([]template.File), args.Error(1)
+}
+func (tm *TemplateManagerMock) ResolverNewPath(oldPath, newDir, lang, workspacePath string) (string, error) {
+	args := tm.Called(oldPath, newDir, lang, workspacePath)
+	return args.String(0), args.Error(1)
+}
+func (tm *TemplateManagerMock) Validate() error {
```
Comment (henriquemoraeszup) [inline internal/mocks/mocks.go:351]: nitpick: function spacing

File: pkg/cmd/create_formula.go [hunk 3]
```diff
@@ -159,52 +166,55 @@ func (c createFormulaCmd) runStdin() CommandRunnerFunc {
...
+		}
+
 		return nil
 	}
 }
 
-func (c createFormulaCmd) create(cf formula.Create) {
-	buildInfo := prompt.Bold("Creating and building formula...")
-	s := spinner.StartNew(buildInfo)
-	time.Sleep(2 * time.Second)
```
Comment (henriquemoraeszup) [inline pkg/cmd/create_formula.go:170]: ❤️

File: pkg/cmd/create_formula_test.go [hunk 4]
```diff
@@ -19,143 +19,312 @@ package cmd
...
+				formulaCreatorMock,
+				templateManagerMock,
+				workspaceMock,
+				inputTextMock,
+				inputTextValidatorMock,
+				inputListMock,
+				tutorialMock,
+				treeMock,
 			)
 			createFormulaCmd.PersistentFlags().Bool("stdin", false, "input by stdin")
```
Comment (henriquemoraeszup) [inline pkg/cmd/create_formula_test.go:169]: can you add a TODO to remove it after deprecation?

File: pkg/cmd/create_formula_test.go [hunk 5]
```diff
@@ -19,143 +19,312 @@ package cmd
...
+	"github.com/ZupIT/ritchie-cli/pkg/api"
+	"github.com/ZupIT/ritchie-cli/pkg/formula"
 	"github.com/ZupIT/ritchie-cli/pkg/formula/creator/template"
-	"github.com/ZupIT/ritchie-cli/pkg/formula/tree"
-	"github.com/ZupIT/ritchie-cli/pkg/prompt"
+	"github.com/ZupIT/ritchie-cli/pkg/rtutorial"
 	"github.com/ZupIT/ritchie-cli/pkg/stream"
 )
 
 func TestNewCreateFormulaCmd(t *testing.T) {
```
Comment (henriquemoraeszup) [inline pkg/cmd/create_formula_test.go:36]: what is the point of this test vs `TestCreateFormulaCmd`?

File: pkg/cmd/create_formula_test.go [hunk 6]
```diff
@@ -19,143 +19,312 @@ package cmd
...
 			in: in{
-				inTextValidator: inputTextValidatorMock{},
-				tm: TemplateManagerCustomMock{
-					ValidateMock: func() error {
-						return nil
-					},
-					LanguagesMock: func() ([]string, error) {
-						return []string{}, errors.New("error on language func")
-					},
-				},
```
Comment (henriquemoraeszup) [inline pkg/cmd/create_formula_test.go:98]: so much simpler, love it

File: pkg/cmd/create_formula_test.go [hunk 7]
```diff
@@ -19,143 +19,312 @@ package cmd
...
+
+			createForm := createFormulaCmd{
+				formula:   creatorMock,
+				workspace: workspaceMock,
+				tutorial:  tutorialMock,
+			}
+
+			got := createForm.create(cf)
+
+			assert.Equal(t, tt.want, got)
```
Comment (henriquemoraeszup) [inline pkg/cmd/create_formula_test.go:283]: can we test the actual files creation?

**Review by henriquemoraeszup — APPROVED** [review]
I am adding this PR to the testify list, nice work

**Pull request merged**